### DISCUSSION TITLE
Vue + Client Routing + Hybrid SPA/SSR

### DISCUSSION THREAD

**Discussion by scottmillionnexient** [vikejs/vike#777] [Discussion]
Are there any sample repos out there that show an implementation with Client Side Routing and hybrid SSR/SPA? Preferably with Vue.

I'm having trouble matching up the official docs with repo examples to achieve what we want. 

- Client Side Routing
- Ability to use SSR or SPA (SSR for login page (unauth users), SPA otherwise. May use SSR for a page or two down the road)
  - Want to prefetch sensitive config data for a login page and save it to Pinia. 

This is pretty close to what we want. Client Side routing. Pinia Store. But only renders pages in SSR. Can't use .client.page. Having trouble modifying this to support SPA (client only rendered).
https://github.com/brillout/vite-plugin-ssr/tree/main/examples/vue-pinia

This is also pretty close to what we want, but w/o Client Side routing. 
https://github.com/AaronBeaudoin/vite-plugin-ssr-example

If what I'm looking for can't be done, please let me know. Appreciate any input or example repos that have Client Side Routing with the ability to use SPA and SSR

**Comment by brillout** [comment]
Yes it's possible, although you'll have to implement it yourself.

I'd recommend to start with Aaron's example, remove the HTML-only part, and then add Pinia.

(FYI, even with the upcoming built-in Vue renderer that includes both SPA and SSR, it's possible that you'll still want to implement your own renderer for complete control over your Pinia integration.)

**Comment by ericarthurc** [comment]
Look at my server and client rendering methods:
https://github.com/Ericarthurc/ericarthurc.com_vite-ssr/tree/main/renderer

This allows for client side routing with ssr and spa hybrid. And allows for navigation over `<a>` anchors between ssr and spa pages without error.

I am using Solid, but hopefully should give you an idea on how to handle this hybrid workflow.
